code/modules/mob/living/simple_animal/hostile/mining_mobs/hivelord.dm

```diff
@@ -909,91 +734,164 @@
+				suit = /obj/item/clothing/suit/space/hardsuit/solgov
+				shoes = /obj/item/clothing/shoes/combat
+				gloves = /obj/item/clothing/gloves/combat
+				mask = /obj/item/clothing/mask/gas/sechailer/swat
+				id = /obj/item/card/id/solgov
+			if(prob(85))
+				back = /obj/item/storage/backpack
+				backpack_contents = list()
+				for(var/i = 1 to 3)
+					if(prob(75))
+						backpack_contents += pickweight(list(
+							/obj/item/reagent_containers/hypospray/medipen/stimpack/traitor = 1,
+							/obj/item/storage/firstaid/tactical = 1,
+							/obj/item/gun/ballistic/automatic/pistol/solgov = 1,
+							/obj/item/gps = 1,
+							/obj/item/stock_parts/cell/gun/upgraded = 2,
+							/obj/item/ammo_box/magazine/pistol556mm = 3,
+							/obj/item/desk_flag/solgov = 3,
+							/obj/item/stack/marker_beacon/ten = 3,
+							/obj/item/detective_scanner = 2,
+							/obj/item/extinguisher/mini = 3,
+							/obj/item/kitchen/knife/combat = 3,
+							/obj/item/flashlight/seclite=3,
+							/obj/item/ammo_casing/shotgun = 3,
+							/obj/item/binoculars = 3,
+							/obj/item/clipboard = 3
+							)
+						)
+			else
+				back = pickweight(list(
+					/obj/item/energyhalberd = 5,
+					/obj/item/gun/ballistic/rocketlauncher/unrestricted = 5
 					)
 				)
+			if(prob(25))
+				belt = /obj/item/storage/belt/military
 			if(prob(50))
-				backpack_contents += pickweight(list(
-					/obj/item/book/granter/spell/knock = 1,
-					/obj/item/book/granter/spell/blind = 1,
-					/obj/item/shadowcloak = 1,
-					/obj/item/book/granter/spell/smoke = 2,
-					/obj/item/reagent_containers/syringe/mulligan = 2,
-					/obj/item/dice/d20/fate/stealth/one_use = 1,
-					/obj/item/clothing/head/chameleon/broken = 3,
-					/obj/item/stack/marker_beacon/ten = 3,
-					/obj/item/grenade/smokebomb = 3,
-					/obj/item/grenade/flashbang = 3
+				r_pocket = pickweight(list(
+					/obj/item/reagent_containers/hypospray/medipen/stimpack = 1,
+					/obj/item/kitchen/knife/letter_opener = 3,
+					/obj/item/radio/off = 3,
+					/obj/item/grenade/syndieminibomb/concussion = 1,
+					/obj/item/melee/transforming/energy/ctf/solgov = 1
+					)
+				)
+			if(prob(70))
+				glasses = pickweight(list(
+					/obj/item/clothing/glasses/sunglasses = 3,
+					/obj/item/clothing/glasses/hud/health = 3,
+					/obj/item/clothing/glasses/hud/health/night = 1,
+					/obj/item/clothing/glasses/night = 2
+					)
+				)
+
+		if("Infiltrator")//WE FINALLY FOUND HIM
+			uniform = /obj/item/clothing/under/syndicate/bloodred
+			gloves = /obj/item/clothing/gloves/color/latex/nitrile/infiltrator
+			suit = /obj/item/clothing/suit/armor/vest/infiltrator
+			head = /obj/item/clothing/head/helmet/infiltrator
+			shoes = /obj/item/clothing/shoes/combat/sneakboots
+			ears = /obj/item/radio/headset
+			if(prob(70))
+				glasses = pickweight(list(
+					/obj/item/clothing/glasses/regular/circle = 1,
+					/obj/item/clothing/glasses/sunglasses = 3,
+					/obj/item/clothing/glasses/thermal/syndi = 2,
```

Comment:
Giving them thermals is a bad idea, regardless of how rare they are.
```suggestion
					/obj/item/clothing/glasses/night = 2,
```

Reply:
We give thermals in the komodo and two ruins, i wouldn't say its too powerful?